pennylane_lightning/src/Gates.hpp

```diff
@@ -137,6 +147,8 @@ namespace Pennylane {
         inline const std::vector<CplxType>& asMatrix() {
             return matrix;
         }
+        virtual void applyGenerator(const StateVector& state, const std::vector<size_t>& indices, const std::vector<size_t>& externalIndices);
```

Comment:
```suggestion
        void applyGenerator(const StateVector& state, const std::vector<size_t>& indices, const std::vector<size_t>& externalIndices);
```
Do we want the virtual here? I think it's for when we are defining base classes.

Reply:
True, just a case of me copying things too quickly without checking.